src/Containers/NutritionFact/NutritionFact.tsx

```diff
@@ -0,0 +1,296 @@
+import React,{useCallback,useMemo,useRef,forwardRef,useEffect,useState} from 'react'
+import styled from 'styled-components'
+import { Mixins } from '../../Utils'
+import {
+    Main,
+    MainInterface,
+} from '../../Utils/BaseStyles';
+import Theme from '../../Themes/ThemeTemplate'
+import {Input} from '../../Inputs/Input/Input'
+
+const EXTRA_PIXEL=1
+const ENTRY_INPUT_WIDTH=20
+const HEADER_ENTRY_INPUT_WIDTH=40
+
+interface IAdditionalProps{
+    ref?:boolean;
+    delay?:boolean;
```

Comment:
What is this used for?

Reply:
this is to set width of the component dynamically. if I do not this then width of the component gets like this:
![Captura de pantalla 2021-03-28 a las 9 23 39](https://user-images.githubusercontent.com/29568460/112745255-51d29e00-8fa7-11eb-837c-b11c2f9582e7.png)
because of the last entry of daily value explication. so to fix that we use delay:true on this last entry and ref:true in the first heading entry, which is the one we want to take width from to set it in the main container as max-width, because in this case the first heading is the one which takes more width

Reply:
Why is it named delay??
Can you put comments directly on the code